src/BlobBasics.java

```diff
@@ -122,31 +109,29 @@ else if (s.getErrorCode().equals("FeatureNotSupportedByEmulator")) {
 
             // Enumerate all containers starting with the prefix "blobbasics-" and list all blobs
             System.out.println("\nEnumerate all containers and starting with the prefix \"blobbasics-\" list all blobs");
-            for (CloudBlobContainer container : blobClient.listContainers("blobbasics-")) {
-                System.out.println(String.format("\tContainer: %s", container.getName()));
-                for (ListBlobItem blob : container.listBlobs()) {
-                    if (blob instanceof CloudBlob) {
-                        System.out.println(String.format("\t\t%s\t: %s", ((CloudBlob) blob).getProperties().getBlobType(), blob.getUri().toString()));
-                    }
-                }
-            }
+            blobServiceClient.listBlobContainers(new ListBlobContainersOptions().setPrefix("blobbasics-"), (Duration)null).forEach(blobContainerItem -> {
+                System.out.println(String.format("\tContainer: %s", blobContainerItem.getName()));
+                BlobContainerClient containerItem = blobServiceClient.getBlobContainerClient(blobContainerItem.getName());
+                containerItem.listBlobs().forEach(blobItem -> {
+                    System.out.println(String.format("\t\t%s\t: %s", blobItem.getProperties().getBlobType(), containerItem.getBlobClient(blobItem.getName()).getBlobUrl()));
+                });
+            });
 
             // Acquire a lease on a container so that another client cannot write to it or delete it
             System.out.println("\nAcquiring a lease on a container to prevent writes and deletes.");
-            container1.acquireLease();
-            System.out.println(String.format("\tSuccessfully acquired a lease on container %s. Lease state: %s.", container1.getName(), container1.getProperties().getLeaseStatus().toString()));
-            container1.breakLease(0);
-            System.out.println(String.format("\tSuccessfully broke the lease on container %s. Lease state: %s.", container1.getName(), container1.getProperties().getLeaseStatus().toString()));
+            AzureBlobStorageImpl azureBlobStorageImpl = new AzureBlobStorageBuilder().pipeline(blobServiceClient.getHttpPipeline()).url(blobServiceClient.getAccountUrl()).build();
```

Comment:
We should be using the lease clients here, right? I'd like to avoid using impl classes in samples.

Reply:
I agree with your proposal, but the existing `BlobContainerClient` does not provide the service interface related to `acquireLease`, so this can only be achieved in this way at this stage.

Reply:
By lease clients I meant the separate [BlobLeaseClient](https://docs.microsoft.com/java/api/com.azure.storage.blob.specialized.blobleaseclient) class used to manage a lease.
```java
BlobLeaseClient leaseClient = new BlobLeaseClientBuilder().containerClient(container1).buildClient();
```

Reply:
The code has been fixed according to your suggestion